src/Microsoft.AspNetCore.Mvc.Core/Controllers/DefaultControllerActivator.cs

```diff
@@ -3,24 +3,22 @@
 
 using System;
 using Microsoft.AspNetCore.Mvc.Core;
-using Microsoft.AspNetCore.Mvc.Internal;
+using Microsoft.AspNetCore.Mvc.Infrastructure;
 
 namespace Microsoft.AspNetCore.Mvc.Controllers
 {
     /// <summary>
     /// <see cref="IControllerActivator"/> that uses type activation to create controllers.
     /// </summary>
-    public class DefaultControllerActivator : IControllerActivator
+    internal sealed class DefaultControllerActivator : IControllerActivator
```

Comment:
We should think about how users extend the framework or what story do we want to tell them. We need to strike a balance between overriding a method or telling them to provide their own entire implementation. (Specially for classes with a lot of dependencies or complex implementations). If people end up copying the code of the internal type they automatically opt-out of patches, etc for that area of the code base, and that's not a good story. In the past we had a lot of problems in traditional web api because of these type of things.